include/gz/math/MecanumDriveOdometry.hh

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_MATH_MECANUMDRIVEODOMETRY_HH_
+#define GZ_MATH_MECANUMDRIVEODOMETRY_HH_
+
+#include <chrono>
+#include <memory>
+#include <gz/math/Angle.hh>
+#include <gz/math/Export.hh>
+#include <gz/math/config.hh>
+
+namespace ignition
+{
+  namespace math
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_MATH_VERSION_NAMESPACE {
+    // Forward declarations.
+    class MecanumDriveOdometryPrivate;
+
+    /// \class MecanumDriveOdometry MecanumDriveOdometry.hh
+    /// gz/math/MecanumDriveOdometry.hh
+    ///
+    /// \brief Computes odometry values based on a set of kinematic
+    /// properties and wheel speeds for a Mecanum-drive vehicle.
+    ///
+    /// A vehicle with a heading of zero degrees has a local
+    /// reference frame according to the diagram below.
+    ///
+    ///       Y
+    ///       ^
+    ///       |
+    ///       |
+    ///       O--->X(forward)
```

Comment:
Can you mention the caveat about wheel radius here? Most users won't see the TODO in the `.cc` file.

Reply:
Done.